pdfs/src/Pdf/FromLatex.hs

```diff
@@ -0,0 +1,130 @@
+
+module Pdf.FromLatex where
+
+escapeInLatex :: Text -> Text
+escapeInLatex text = "\\verb|" <> {-Text.filter (\= '|')-} text <> "|"
+
+type Locale = Text
+
+data Date = Date
+    { year  :: Int
+    , month :: Int
+    , day   :: Int
+    }
+
+dateToString :: Locale -> Date -> Text
+dateToString _ (Date y m d) = show d <> "." <> show m <> "." <> show y
+
+data StudentInfo = StudentInfo
+    { siName        :: Text
+    , siDateOfBirth :: Date
+    , siSections    :: [Section]
+    }
+
+data Section = Section
+    { sEducator :: Text
+    , sDiploma  :: Diploma
+    , sCourses  :: [Course]
+    }
+
+data Diploma = Diploma
+    { dEducatorUrl     :: Text
+    , dEducationPeriod :: (Int, Int)
+    , dDiplomaId       :: Text
+    , dDegreeLevel     :: Text
+    , dMajor           :: Text
+    , dMinor           :: Text
+    , dPartTime        :: Bool
+    }
+
+data Course = Course
+    { cName        :: Text
+    , cLanguage    :: Text
+    , cHours       :: Int
+    , cEstcCredits :: Maybe Int
+    , cResult      :: Text
+    }
+
+generate :: Locale -> StudentInfo -> Text
```

Comment:
Consider using `Builder` instead of `Text`.

Reply:
Since `<>` for `Text` is `O(n)` memory allocation and takes `O(n)` time (resulting in O(n^2) to build a whole certificate), while for `Builder` append requires `O(1)` memory and `O(1)` time.